spark.md

```diff
@@ -167,6 +170,52 @@ Note that ```${scala.binary.version}``` is a Maven property with the value ```2.
 
 The [Deeplearning4j examples repo](https://github.com/deeplearning4j/dl4j-examples) ([old examples here](https://github.com/deeplearning4j/dl4j-spark-cdh5-examples)) contains a number of Spark examples.
 
+## <a name="gpusspark">Training with GPUs on Spark</a>
+
+There are some issues to be aware of when running Deeplearning4j with CUDA on Spark, with cluster managers such as YARN.
+
+
+### <a name="yarngpus">YARN and GPUs</a>
+
+[YARN](https://hadoop.apache.org/docs/current/hadoop-yarn/hadoop-yarn-site/YARN.html) is a commonly used cluster resource management/scheduling tool for Hadoop clusters. Running Deeplearning4j with GPUs on Spark+YARN is possible, but (unlike memory and CPU resources) YARN does not track/handle GPUs as a resource. Consequently, some additional steps are typically required to run DL4J with GPUs on YARN. Note however that this is more of an issue with heterogeneous-hardware clusters (i.e., some machines with GPUs, others without), as YARN won't natively allocate your jobs only to machines that have GPUs.
```

Comment:
> Note however that this is more of an issue with heterogeneous-hardware clusters (i.e., some machines with GPUs, others without),

Not necessarily : let's assume you have one GPU on every machine, and a workload that does use it. If you do schedule with a native YARN, the major risk is that based on memory + CPU calculations, the framework allocates 2 distinct jobs (from different data scientists, say) on the same executor : they would overstep each other in terms of GPU memory requirement.

This is usually solved by "faking" the requirements of the YARN job so that cpu-and-memory-wise, the job, when scheduled, reserves the memory of the whole executor it is running on. Another way is to run executors in containers (where those containers [declare and reserve their GPU usage](https://github.com/NVIDIA/nvidia-docker/wiki/GPU-isolation)).